scmgr/app.go

```diff
@@ -0,0 +1,562 @@
+			log.Info("  " + sb.Sprint(short))
+		}
+	}
+	return nil
+}
+
+// lsIndex writes one index-file for every known skipchain and an index.html
+// for all skiplchains.
+func lsIndex(c *cli.Context) error {
+	output := c.Args().First()
+	if len(output) == 0 {
+		return errors.New("Missing output path")
+	}
+
+	cleanHTMLFiles(output)
+
+	cfg, err := loadConfig(c)
+	if err != nil {
+		return errors.New("couldn't read config: " + err.Error())
+	}
+
+	// Get the list of genesis block
+	genesis := sbl{}
+	for _, sb := range cfg.Sbm.SkipBlocks {
+		if sb.Index == 0 {
+			genesis = append(genesis, sb)
+		}
+	}
+
+	sort.Sort(genesis)
+
+	// Build the json structure
+	blocks := jsonBlockList{}
+	blocks.Blocks = make([]jsonBlock, len(genesis))
+	for i, g := range genesis {
+		block := &blocks.Blocks[i]
+		block.GenesisID = hex.EncodeToString(g.Hash)
+		block.Servers = make([]string, len(g.Roster.List))
+		block.Data = g.Data
+
+		for j, server := range g.Roster.List {
+			block.Servers[j] = server.Address.Host() + ":" + server.Address.Port()
+		}
+
+		// Write the genesis block file
+		content, _ := json.Marshal(block)
+		err := ioutil.WriteFile(filepath.Join(output, block.GenesisID+".html"), content, 0644)
+
+		if err != nil {
+			log.Info("Cannot write block-specific file")
+		}
+	}
+
+	content, err := json.Marshal(blocks)
+	if err != nil {
+		log.Info("Cannot convert to json")
+	}
+
+	// Write the json into the index.html
+	err = ioutil.WriteFile(filepath.Join(output, "index.html"), content, 0644)
+	if err != nil {
+		log.Info("Cannot write in the file")
+	}
+
+	return nil
+}
+
+func lsFetch(c *cli.Context) error {
+	cfg := getConfigOrFail(c)
+	rec := c.Bool("recursive")
+	sisAll := map[network.ServerIdentityID]*network.ServerIdentity{}
+	group := readGroup(c, 0)
+	for _, sb := range cfg.Sbm.SkipBlocks {
+		for _, si := range sb.Roster.List {
+			sisAll[si.ID] = si
+		}
+	}
+	for _, si := range group.Roster.List {
+		sisAll[si.ID] = si
+	}
+	log.Info("The following ips will be searched:")
+	for _, si := range sisAll {
+		log.Info(si.Address)
+	}
+	client := skipchain.NewClient()
+	sisNew := sisAll
+	for len(sisNew) > 0 {
+		log.Print("sisNew is", sisNew)
+		sisIterate := sisNew
+		sisNew = map[network.ServerIdentityID]*network.ServerIdentity{}
+		for _, si := range sisIterate {
+			log.Info("Fetching all skipchains from", si.Address)
+			gasr, cerr := client.GetAllSkipchains(si)
+			if cerr != nil {
+				// Error is not fatal here - perhaps the node is down,
+				// but we can continue anyway.
+				log.Error(cerr)
+				continue
+			}
+			for _, sb := range gasr.SkipChains {
+				log.Printf("Found skipchain %x", sb.SkipChainID())
+				cfg.Sbm.SkipBlocks[string(sb.SkipChainID())] = sb
+				if rec {
+					log.Print("rec")
+					for _, si := range sb.Roster.List {
+						if _, exists := sisAll[si.ID]; !exists {
+							log.Print("Adding", si)
+							sisNew[si.ID] = si
+							sisAll[si.ID] = si
+						}
+					}
+				}
+			}
+		}
+	}
```

Comment:
All of the above brings quite some conditional complexity -> 6 inner conditions. It'd be best if it were reduced to 2 or 3, but if it does not make sense (i.e. it only adds new functions only being used here), then it's fine.